perma_web/api/tests/utils.py

```diff
@@ -135,19 +135,12 @@ def setUp(self):
         super(ApiResourceTestCaseMixin, self).setUp()
 
         self.api_client = LoggingAPIClient()
-
```

Comment:
I think we still want to run individual tests in individual temp dirs, right? Ideally no state gets preserved from one test to another. The thing in `fab` would be in addition to this instead of replace it.

Reply:
"Ideally no state gets preserved from one test to another." Totally! Unfortunately, that isn't reliably achieved by the setup/teardown code that was here.

It's super interesting to watch go by in pdb: the temp directories are indeed created, settings.MEDIA_ROOT is reset appropriately at the start and finish of each test, and the tmp directory and is deleted. But, unless it's the first test that has been run that attempts to access default_storage, these steps have no effect on where the files are actually saved. 

On develop: if the perma app tests are run first, files are saved in whatever directory MEDIA_ROOT was, when storage was first accessed there. If the api tests are run first, files are saved in whatever tmp directory was created by the first api test. To watch this happen, put a breakpoint as the first line of the api tests tearDown.
```
> fab test:apps='api.tests.test_link_resource'

[localhost] local: coverage run manage.py test --settings perma.settings.deployments.settings_testing api.tests.test_link_resource

Creating test database for alias 'default'...
Creating test database for alias 'perma-cdxline'...

<truncating! press c until you see a test run that creates a capture>

ET5Z-6VYD step 0: Starting capture
....
ET5Z-6VYD capture succeeded.
> /vagrant/perma_web/api/tests/utils.py(151)tearDown()
-> settings.MEDIA_ROOT = self._media_org
(Pdb) import os
(Pdb) self._media_tmp
'/tmp/tmpOWPyIu'
(Pdb) os.listdir(self._media_tmp)
['warcs']
(Pdb) n
> /vagrant/perma_web/api/tests/utils.py(152)tearDown()
-> shutil.rmtree(self._media_tmp)
(Pdb) c

<truncating again! press c until you see another capture created>

28ZC-B2GF step 0: Starting capture
...
28ZC-B2GF capture succeeded.
> /vagrant/perma_web/api/tests/utils.py(151)tearDown()
-> settings.MEDIA_ROOT = self._media_org
(Pdb) self._media_tmp
'/tmp/tmpYIeYlg'
(Pdb) os.listdir(self._media_tmp)
[]
(Pdb) os.listdir('/tmp/tmpOWPyIu')
['warcs']
```

So, I thought it was best to remove these setUp and tearDown methods, until we can figure out a way to get them to work as advertised. Googling has not suggested a solution: the best anybody seems to have come up with is writing a customer test runner that, when setting up the testing environment, does the same thing I do here via fab.

Maybe we can manipulate the location attribute of the FileStorage object directly (if we can figure out how to get a hold of the instance in use...) https://docs.djangoproject.com/en/1.11/_modules/django/core/files/storage/#FileSystemStorage.